core/amber/src/main/scala/edu/uci/ics/amber/engine/common/storage/DatasetFileDocument.scala

```diff
@@ -53,9 +63,15 @@ class DatasetFileDocument(fileFullPath: Path) extends VirtualDocument[Nothing] {
   }
 
   override def remove(): Unit = {
+    // first remove the temporary file
     tempFile match {
       case Some(file) => Files.delete(file.toPath)
       case None       => // Do nothing
     }
+    // then remove the dataset file
+    GitVersionControlLocalFileStorage.removeFileFromRepo(
+      PathUtils.getDatasetPath(UInteger.valueOf(did)),
```

Comment:
those two information are kind of redundant (the datasetPath is provided twice. You can fix it later.)

Reply:
I will do it in future PR